### ISSUE TITLE
from_pretrained does not load the modified part of model 

### ISSUE THREAD

**Issue by ghost** [huggingface/transformers#9307]
## Environment info

     
- `transformers` version:
- Platform: Linux
- Python version: 3.7
- PyTorch version (GPU?): 1.7
- Tensorflow version (GPU?): - 
- Using GPU in script?: no 
- Using distributed or parallel set-up in script?: no 

### Who can help

 Model Cards: @julien-c
## Information

Hi
1) I am observing that if one modifies a model lest say T5ForConditionalGeneration and then use  T5ForConditionalGeneration.from_pretrained(...) then not all components of the models are loaded, meaning that the 
parts of the model the user has modified are initialized to random! 

2) I observe this from accuracy, could you tell me how I can check which weights from_pretrained is loading, I am a bit lost in the repository. thanks
## Expected behavior

If the model has been changed to have more layers, .... all the trained weights need to be loaded.

**Comment by juliahane** [comment]
In the documentation, this is written how from_pretrained works for untouched models, but I cannot see how this works when one modifies the model.

**Comment by patrickvonplaten** [comment]
Hey @juliahane,

could you please provide a code snippet showcasing the unintended behavior? Thanks!
